src/Indexation/AbstractType.php

```diff
@@ -54,7 +56,13 @@ public function update($object, $id)
             throw new NoElasticaTypeAvailable();
         }
 
-        $this->type->updateDocument(new Document($id, $object));
+        try {
+            $this->type->updateDocument(new Document($id, $object));
+        } catch (ResponseException $exception) {
+            if ('document_missing_exception' === $exception->getResponse()->getFullError()['type']) {
+                throw new DocumentNotFoundException();
```

Comment:
Please add the `ResponseException` as previous exception (last parameter of the constructor). And add also a nice error message :) .